### ISSUE TITLE
Examples of Service Containers do not work

### ISSUE THREAD

**Issue by patvin80** [MicrosoftDocs/azure-devops-docs#4187]
I picked the sample above and pasted it in a pipeline, but was not able to execute the pipeline. The error I get when I use the nginx and redis pipeline is when the apt get command is run. 

[command]/bin/bash --noprofile --norc /home/vsts/work/_temp/683b6fb6-dbe3-4dea-a205-d102ee22e8bf.sh

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
/home/vsts/work/_temp/683b6fb6-dbe3-4dea-a205-d102ee22e8bf.sh: line 2: psql: command not found

On changing the command to run as sudo, the challenge is Ubuntu 16 onwards does not have sudo installed and since apt-get does not work cannot install it either. As a workaround I am using the vsts ubuntu 14.04 image but that increases the build time. 

The documentation needs to be corrected or please identify if I am doing something wrong.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 7d1c794c-d676-fc7e-758e-3464f16a1554
* Version Independent ID: 941de04a-2d34-ec82-8612-00e4070de673
* Content: [Service Containers - Azure Pipelines & TFS](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/service-containers?view=azure-devops&tabs=yaml#feedback)
* Content Source: [docs/pipelines/process/service-containers.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/pipelines/process/service-containers.md)
* Product: **devops**
* Technology: **devops-cicd**
* GitHub Login: @thomps23
* Microsoft Alias: **dakale**

**Comment by andyjlewis** [comment]
@thomps23 please assign to @dakale

**Comment by wlongxiang** [comment]
I got the same problem here

**Comment by masaori** [comment]
It seems that I can define 'services' only under 'jobs' according to the document about YAML schema ( https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#server )
So I think those examples in this article don't work in the first place. (Actually, it didn't on my env.)

How did you guys write your .yml?

**Comment by azizaitali** [comment]
The following didn't work for me:

resources:
  containers:
  - container: sql
    image: mcr.microsoft.com/mssql/server
    ports:
    - 8080:80
    env:
        SA_PASSWORD: "*****"
        ACCEPT_EULA: "Y"
  - container: api
    image: my-api:latest
    ports:
    - 8080:80
    env:
      connectionString: "Server=tcp:sql,1433;Initial Catalog=my-db;Persist Security Info=False;User ID=sa;Password=*****;MultipleActiveResultSets=False;Connection Timeout=30;"
services:
  sql: sql
  api: api
pool:
  vmImage: 'ubuntu-16.04'
steps:
- script: |
    ping localhost -c 10
    curl localhost:8080

**Comment by orionstudt** [comment]
The container job example does not work because the ubuntu container is not running as `root` user so you can't run `apt-get`. It would be nice if the examples were valid

**Comment by cxwtool** [comment]
This issue hasn't been updated in more than 180 days, so we've closed it. If you feel the issue is still relevant and needs fixed, please reopen it and we'll take another look. We appreciate your feedback and apologize for any inconvenience.

**Comment by 3noch** [comment]
The documentation is still wrong.

**Comment by Atheuz** [comment]
I also ran into an issue where because Azure Pipelines was not running the service container as root, it couldn't install anything through apt-get. 

I had to modify the base image before using it in the pipeline, which is really clunky. 

Why is it the case that service containers are not permitted to run as root? It really restricts what they are able to do.

The documentation should also be updated to reflect that service containers may not do what you want, because they're not running as root.
